wagtail/images/templates/wagtailimages/images/index.html

```diff
@@ -31,13 +32,31 @@
     {% endif %}
 
     <div class="nice-padding">
-        {% if collections %}
-            <form class="image-search search-bar" action="{% url 'wagtailimages:index' %}" method="GET" novalidate>
-                <ul class="fields">
+        <form class="image-search search-bar" action="{% url 'wagtailimages:index' %}" method="GET" novalidate>
+            <ul class="fields">
+                {% if collections %}
                     {% include "wagtailadmin/shared/collection_chooser.html" %}
-                </ul>
-            </form>
-        {% endif %}
+                    {% if current_tag %}
+                        <input type="hidden" name="tag" value="{{ current_tag }}" />
+                    {% endif %}
+                {% endif %}
+                {% if popular_tags %}
+                <li class="taglist">
+                    <h3>{% trans 'Popular tags' %}</h3>
```

Comment:
I think `h3` is probably not suitable here, as there is no other h3 on the page and the nearest h2 is in the menu.

This should probably be something like
`<label for="">{% trans 'Popular Tags:' %}</label>`

Notes;
* the for will need to go somewhere, perhaps to the 'selected' tag (using an id like `current-tag`) as you cannot reference a hidden field with a label's for attribute
* go with title case and colon `:` at the end so this matches the `Collections:` from above
* Might need to do some style tweaks so it looks the same as the `Collections:` text
* A benefit of using `label` is that it will be display block, so it will wrap nicer on small devices and again look similar to the field above

Reply:
I tried to keep it close to how it's done in [chooser.html](https://github.com/wagtail/wagtail/blob/eaad01308142337739d7edd9f08bd8e714be23f0/wagtail/images/templates/wagtailimages/chooser/chooser.html#L28), that's why I chose an `h3` tag.
It probably makes sense, to change it there too. What do you think?

Anyway the chooser.html template is also missing a `Clear choice` button. First, I'd like to finish what I started here, before I start working on other places.

Reply:
As to [MDN](https://developer.mozilla.org/de/docs/Web/HTML/Element/label#Attributes) the labels for attribute should reference the id of a [labelable](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#Form_labelable) form-related element ( `<button>`, `<input>`, `<keygen>`, `<meter>`, `<output>`, `<progress>`, `<select>`, and `<textarea>`).

I think a `fieldset` with a `legend` element would be the proper way to go. What do you think?

Reply:
@Hopiu - good investigation, I had not dug that deep into label references :) - I agree that a fieldset would make sense as we are using <a/> as buttons here (hopefully we will fix that in the future as a stand alone issue).

And yeah - it would be nice to align with the chooser here but they are in different contexts and we want to try and work with the markup/design/interaction already present on this page I reckon. plus h3 may not be suitable there either, but that can be done in a future improvement haha.